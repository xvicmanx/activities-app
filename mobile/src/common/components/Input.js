import React, { forwardRef } from 'react';
import { StyleSheet } from 'react-native';
import { Input as InputRNE } from 'react-native-elements';
import { COLORS } from '../../constants';

const Input = forwardRef((props, ref) => {
  const { iconName, error, onChange, disable, ...rest } = props;

  return (
    <InputRNE
      leftIcon={{ type: 'ionicon', name: iconName, color: COLORS.primary }}
      errorStyle={styles.error}
      errorMessage={error}
      onChangeText={onChange}
      editable={!disable}
      inputStyle={styles.input}
      inputContainerStyle={error && styles.inputContainerError}
      ref={ref}
      {...rest}
    />
  );
});

const styles = StyleSheet.create({
  error: {
    color: COLORS.danger,
  },
  success: {
    color: COLORS.green,
  },
  input: {
    color: COLORS.primary,
    fontSize: 15,
  },
  inputContainerError: {
    borderColor: COLORS.danger,
  },
});

export default Input;
