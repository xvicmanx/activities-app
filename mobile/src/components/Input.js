import React, { forwardRef } from 'react';
import { Input as InputRNE } from 'react-native-elements';
import { COLORS } from '../constants';

const Input = forwardRef((props, ref) => {
  const { iconName, error, onChange, disable, ...rest } = props;

  return (
    <InputRNE
      leftIcon={{ type: 'ionicon', name: iconName, color: COLORS.primary }}
      errorStyle={{ color: COLORS.danger }}
      errorMessage={error}
      onChangeText={onChange}
      editable={!disable}
      inputStyle={{ color: COLORS.primary, fontSize: 15 }}
      ref={ref}
      {...rest}
    />
  );
});

export default Input;
