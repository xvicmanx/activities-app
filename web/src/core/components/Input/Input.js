import React from 'react';
import { Form } from 'react-bulma-components';
import './Input.css';

const Input = ({ className, ...rest }) => {
  return (
    <Form.Control className="Control">
      <Form.Input
        autoComplete="off"
        className={`Input ${className}`}
        {...rest}
      />
    </Form.Control>
  );
};

export default Input;
