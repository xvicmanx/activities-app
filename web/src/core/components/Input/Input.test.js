import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('Input', () => {
  it('renders properly', () => {
    const component = renderer.create(
      <Input className="test" value="foo" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});