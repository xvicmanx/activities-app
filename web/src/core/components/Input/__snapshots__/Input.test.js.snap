// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders properly 1`] = `
<Control
  className="Control"
  renderAs="div"
>
  <Input
    autoComplete="off"
    className="Input test"
    disabled={false}
    isStatic={false}
    placeholder=""
    readOnly={false}
    type="text"
    value="foo"
  />
</Control>
`;
